rust/deny.toml

```diff
@@ -122,6 +122,7 @@ allow = [
   "BSL-1.0",
   "Zlib",
   "CDLA-Permissive-2.0",
+  "NCSA",
```

Comment:
Hm, do we need to add this license to https://github.com/firezone/firezone/blob/main/docs/NOTICE.txt?

Reply:
I'll approve but I need to look into that. I believe we are supposed to be. Perhaps there's an automated system for this.